src/map/map.cpp

```diff
@@ -4319,37 +4319,493 @@ void map_skill_damage_free(struct map_data *m) {
  * @param other Rate to Other target
  * @param caster Caster type
  **/
-void map_skill_damage_add(struct map_data *m, uint16 skill_id, int pc, int mob, int boss, int other, uint8 caster) {
+void map_skill_damage_add(struct map_data *m, uint16 skill_id, int rate[SKILLDMG_MAX], uint16 caster) {
```

Comment:
Array size has no effect in this context. It will be converted into a pointer anyways.